content/learning-paths/servers-and-cloud-computing/onnx-on-azure/benchmarking.md

```diff
@@ -0,0 +1,137 @@
+You can also compare the inference time between Cobalt 100 (Arm64) and similar D-series x86_64-based VMs on Azure.
+As noted before, the steps to benchmark remain the same, whether it's a Docker container or a custom VM.
+
+## Run the performance tests using onnxruntime_perf_test
+The onnxruntime_perf_test is a performance benchmarking tool included in the ONNX Runtime source code. It is used to measure the inference performance of ONNX models under various runtime conditions (like CPU, GPU, or other execution providers). 
+
+### Install Required Build Tools
+
+```console
+$ tdnf install -y cmake make gcc-c++ git
+```
+### Try Installing protobuf from Package
+
+```console
+$ tdnf install -y protobuf protobuf-devel
+```
+Then verify:
+```console
+$ protoc --version
+```
+You should see something like:
+
+```output
+libprotoc 3.X.X
+```
+If the command fails or the version is too old for ONNX Runtime, proceed with building from source.
+
+###  Build Protobuf from Source (Safe on Azure Linux)
+
+```console
+$ git clone --branch v21.12 https://github.com/protocolbuffers/protobuf.git
+$ cd protobuf
+$ git submodule update --init --recursive
+$ mkdir build && cd build
+$ cmake .. -DCMAKE_BUILD_TYPE=Release
+$ make -j$(nproc)
+$ sudo make install
+$ sudo ldconfig
+```
+Then verify:
+```console
+$ protoc --version
+```
+
+output:
+```output
+libprotoc 25.3
+```
+You should now see the correct version (e.g., libprotoc 25.).
+
+### Clone and Build ONNX Runtime from Source:
+
+{{% notice Note %}}
+onnxruntime_perf_test isn’t available as a pre-built binary artifact for any platform. So, you have to build it from the source, which is expected to take around 40-50 minutes. 
+{{% /notice %}}
+
+```console
+$ tdnf install -y protobuf-compiler libprotobuf-dev libprotoc-dev 
+$ git clone --recursive https://github.com/microsoft/onnxruntime 
+$ cd onnxruntime
+```
+Now, build the benchmark by using below command:
+
+```console
+./build.sh --config Release --build_dir build/Linux --build_shared_lib --parallel --build --update --skip_tests 
+```
+output:
+```output
+[100%] Building CXX object CMakeFiles/onnxruntime_test_all.dir/home/azureuser/new/onnxruntime/onnxruntime/test/lora/lora_test.
+[100%] Building CXX object CMakeFiles/onnxruntime_test_all.dir/home/azureuser/new/onnxruntime/onnxruntime/test/unittest_main/tain.cc.o
+[100%] Linking CXX executable onnxruntime_test_all
+[100%] Built target onnxruntime_test_all
+2025-07-09 07:09:04,831 build [INFO] - Build complete
+```
+This confirms that the onnx benchmarking is equally available to be used on Linux based platforms. 
+
+### Run the benchmark
+To run ONNX benchmarking on Azure virtual machines or containers, use the same scripts and methodology across both CPU architectures:
```

Comment:
Now that the benchmarking tool has been built, you can benchmark the **squeezenet-int8.onnx** model, as below: